test/jest/utils.spec.ts

```diff
@@ -0,0 +1,130 @@
+import { queryStringParser, getCookies, getHref } from '../../src/utils';
+import { deleteAllCookies } from './utils';
+
+
+describe('Utils', () => {
+    describe('getCookies', () => {
+        beforeEach(() => {
+            window.document.cookie = '_cookie1=1234';
+            window.document.cookie = '_cookie2=39895811.9165333198';
+            window.document.cookie = 'foo=bar';
+            window.document.cookie = 'baz=qux';
+        });
+
+        afterEach(() => {
+            deleteAllCookies();
+        });
+
+        it('returns all cookies as an object', () => {
+
+            const expectedResult = {
+                foo: 'bar',
+                '_cookie1': '1234',
+                '_cookie2': '39895811.9165333198',
+                baz: 'qux',
+            };
+
+            expect(getCookies()).toEqual(expectedResult);
+        });
+
+        it('returns only the cookies that match the keys', () => {
+            const expectedResult = {
+                foo: 'bar',
+                baz: 'qux',
+            };
+
+            expect(getCookies(['foo', 'baz'])).toEqual(expectedResult);
+        });
+
+        it('returns an empty object if no keys are found', () => {
+            expect(getCookies(['quux', 'corge'])).toEqual({});
+        });
+
+        it('returns an empty object if there are no cookies', () => {
+            deleteAllCookies();
+
+            expect(getCookies()).toEqual({});
+        });
+
+        it('returns an empty object if window is undefined', () => {
+            const originalWindow = global.window;
+            delete global.window;
+
+            expect(getCookies()).toEqual({});
+
+            global.window = originalWindow
+        });
+    });
+
+    describe('queryyStringParser', () => {
```

Comment:
can you add 2 more tests (I think these are true statements):
1. returns an empty object if there are no query parameters
2. returns an object with all query params if no keys are passed